dm/tests/all_mode/run.sh

```diff
@@ -320,6 +320,56 @@ function test_expression_filter() {
 	echo "[$(date)] <<<<<< finish test_expression_filter >>>>>>"
 }
 
+function test_regexpr_router() {
+	echo "[$(date)] <<<<<< start test_regexpr_router >>>>>>"
+	cleanup_process
+	run_sql_file $cur/data/db1.regexpr.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1
+	run_sql_file $cur/data/db2.regexpr.sql $MYSQL_HOST2 $MYSQL_PORT2 $MYSQL_PASSWORD2
+	run_sql_tidb 'drop database if exists dtest3;'
+	run_sql_tidb 'create database dtest3;'
+	run_sql_tidb 'drop database if exists dtest4;'
+	run_sql_tidb 'create database dtest4;'
+	run_sql_tidb 'create table if not exists dtest3.dtable3(a int, b int);'
+	run_sql_tidb 'create table if not exists dtest4.dtable4(a int, b int);'
+	# start DM worker and master
+	run_dm_master $WORK_DIR/master $MASTER_PORT $cur/conf/dm-master.toml
+	check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT
+	check_metric $MASTER_PORT 'start_leader_counter' 3 0 2
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+	run_dm_worker $WORK_DIR/worker2 $WORKER2_PORT $cur/conf/dm-worker2.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER2_PORT
+	# operate mysql config to worker
+	cp $cur/conf/source1.yaml $WORK_DIR/source1.yaml
+	cp $cur/conf/source2.yaml $WORK_DIR/source2.yaml
+	sed -i "/relay-binlog-name/i\relay-dir: $WORK_DIR/worker1/relay_log" $WORK_DIR/source1.yaml
+	sed -i "/relay-binlog-name/i\relay-dir: $WORK_DIR/worker2/relay_log" $WORK_DIR/source2.yaml
+	dmctl_operate_source create $WORK_DIR/source1.yaml $SOURCE_ID1
+	dmctl_operate_source create $WORK_DIR/source2.yaml $SOURCE_ID2
+	cp $cur/conf/regexpr-task.yaml $WORK_DIR/regexpr-task.yaml
+	sed -i "s/name: test/name: $ILLEGAL_CHAR_NAME/g" $WORK_DIR/regexpr-task.yaml
+
+	# error config
+	# there should be a error message like "Incorrect argument type to variable 'tidb_retry_limit'"
+	# but different TiDB version output different message. so we only roughly match here
+	sed -i 's/tidb_retry_limit: "10"/tidb_retry_limit: "fjs"/g' $WORK_DIR/regexpr-task.yaml
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"start-task $WORK_DIR/regexpr-task.yaml --remove-meta" \
+		"tidb_retry_limit" 1
+
+	sed -i 's/tidb_retry_limit: "fjs"/tidb_retry_limit: "10"/g' $WORK_DIR/regexpr-task.yaml
+	dmctl_start_task "$WORK_DIR/regexpr-task.yaml" "--remove-meta"
+
+	check_sync_diff $WORK_DIR $cur/conf/regexpr_diff_config.toml
+
+	cleanup_data test3gd8917777
+	cleanup_data test4s_2022
+	cleanup_data_upstream test3gd8917777
+	cleanup_data_upstream test4s_2022
+	cleanup_process
```

Comment:
i think call cleanup_data(log) before cleanup_process, because if your process is hung, you can not see logs any more

Reply:
Would u mind getting me some hint about the `cleanup_data(log)` since I couldn't find such a function. In addition, cleaning the data and then killing the processes is strictly followed the scheme of other tests in that script. Could you describe it in more detail cuz I've never met the situation that I couldn't see the logs. Thx in advance.

Reply:
p.s. even the some tests fail, the logs remain in the /tmp/dm_test without being broken.

Reply:
cleanup_data is [here](https://github.com/pingcap/tiflow/blob/239a21fe146a2b2950dad06e7d4e201791d4d030/dm/tests/_utils/test_prepare#L4)

>  is strictly followed the scheme of other tests 

yes i used to write test like this unitl i found if dm-worker never exit(some bug code with my local dev branch), i can't see the log of dm-worker because whole test dir is deleted. so if there is **no bug** in our code, call cleanup_data before kill process is ok 😂

Reply:
Yes, deleting the data before killing the processes will make the error logs invisible if the error occurs when exiting the processes😂. But I don't think that is the case we need to test in the integration test😂

Reply:
we need not test this case, i just say call cleanup_data after kill process is good for debug 😬